CRM/Member/Form/Membership.php

```diff
@@ -1922,4 +1922,86 @@ protected function processContribution(
+    $recurParams['financial_type_id'] = $params['financial_type_id'] ?? NULL;
+    $recurParams['currency'] = $params['currency'] ?? NULL;
+    $recurParams['payment_instrument_id'] = $params['payment_instrument_id'];
+
+    // CRM-14354: For an auto-renewing membership with an additional contribution,
+    // if separate payments is not enabled, make sure only the membership fee recurs
+    if (!empty($form->_membershipBlock)
+      && $form->_membershipBlock['is_separate_payment'] === '0'
+      && isset($params['selectMembership'])
+      && $form->_values['is_allow_other_amount'] == '1'
+      // CRM-16331
+      && !empty($form->_membershipTypeValues)
+      && !empty($form->_membershipTypeValues[$params['selectMembership']]['minimum_fee'])
+    ) {
+      $recurParams['amount'] = $form->_membershipTypeValues[$params['selectMembership']]['minimum_fee'];
+    }
+
+    $recurParams['is_test'] = 0;
+    if (($form->_action & CRM_Core_Action::PREVIEW) ||
+      (isset($form->_mode) && ($form->_mode == 'test'))
+    ) {
+      $recurParams['is_test'] = 1;
+    }
+
+    $recurParams['start_date'] = $recurParams['create_date'] = $recurParams['modified_date'] = date('YmdHis');
+    if (!empty($params['receive_date'])) {
+      $recurParams['start_date'] = date('YmdHis', strtotime($params['receive_date']));
+    }
+    $recurParams['invoice_id'] = $params['invoiceID'] ?? NULL;
+    $recurParams['contribution_status_id'] = CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Pending');
+    $recurParams['payment_processor_id'] = $params['payment_processor_id'] ?? NULL;
+    $recurParams['is_email_receipt'] = $params['is_email_receipt'] ?? NULL;
+    // we need to add a unique trxn_id to avoid a unique key error
+    // in paypal IPN we reset this when paypal sends us the real trxn id, CRM-2991
+    $recurParams['trxn_id'] = $params['trxn_id'] ?? $params['invoiceID'];
+    $recurParams['financial_type_id'] = $contributionType->id;
+
+    $campaignId = $params['campaign_id'] ?? $form->_values['campaign_id'] ?? NULL;
+    $recurParams['campaign_id'] = $campaignId;
+    $recurring = CRM_Contribute_BAO_ContributionRecur::add($recurParams);
+    if (is_a($recurring, 'CRM_Core_Error')) {
+      CRM_Core_Error::displaySessionError($recurring);
+      $urlString = 'civicrm/contribute/transact';
+      $urlParams = '_qf_Main_display=true';
+      if (get_class($form) == 'CRM_Contribute_Form_Contribution') {
+        $urlString = 'civicrm/contact/view/contribution';
+        $urlParams = "action=add&cid={$form->_contactID}";
+        if ($form->_mode) {
+          $urlParams .= "&mode={$form->_mode}";
+        }
+      }
+      CRM_Utils_System::redirect(CRM_Utils_System::url($urlString, $urlParams));
+    }
+    // Only set contribution recur ID for contributions since offline membership recur payments are handled somewhere else.
+    if (!is_a($form, "CRM_Member_Form_Membership")) {
```

Comment:
@eileenmcnaughton any reason to not get rid of this now?

Reply:
I figured I'd remove it once moved  - I think most of the lines above can go too - but since you've looked I'll push that up

Reply:
@seamuslee001 I added a commit that removes the bulk of the unrelated code

Reply:
thanks makes more sense in this shift IMO MOPed it